flake-module.nix

```
@@ -0,0 +1,85 @@
+{ treefmtLib }:
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    types;
+in
+{
+  _file = __curPos.file;
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.treefmt = mkOption {
+          description = ''
+            treefmt flake module options
+          '';
+          type = types.submodule {
+            options = {
+              config = mkOption {
+                description = "treefmt-nix configuration";
+                type = types.raw; # TODO: Should use module-options.nix
+              };
+              module = mkOption {
+                type = types.raw; # TODO: module type?
+                description = "Evaluated module of treefmt-nix";
+                default =
+                  treefmtLib.evalModule pkgs config.treefmt.config;
+              };
+              programs = mkOption {
+                type = types.attrsOf types.package;
+                description = "Attrset of formatter programs enabled in configuration";
+                default =
+                  pkgs.lib.concatMapAttrs
+                    (k: v:
+                      if v.enable
+                      then { "${k}" = v.package; }
+                      else { })
+                    config.treefmt.module.config.programs;
+              };
+              wrapper = mkOption {
+                type = types.package;
+                description = "Wrapper treefmt using the current configuration";
+                default = config.treefmt.module.config.build.wrapper;
+              };
+            };
+          };
+        };
+      });
+  };
+  config = {
+    perSystem = { config, self', inputs', pkgs, ... }: {
```

Everything happens in the `perSystem` module, so you could simplify this to more or less:

```suggestion
  config = {
```

It's a bit of a weird (and incomplete) diff, but the idea is to merge both submodules.